### ISSUE TITLE
Be more permissive on boto3 version

### ISSUE THREAD

**Issue by yonigottesman** [aio-libs/aiobotocore#1028]
Currently the boto3 version is
`'boto3': ['boto3>=1.26.161,<1.26.162']`
This causes lots of conflicts with other packages (in our case sagemaker)
Can the versions be something like
`'boto3': ['boto3>=1.26.161,<1.27.0']`?

**Comment by its-felix** [comment]
Second this.

Also, it would be helpful if there would be a `WARN` log if the actual versions don't match the compatible range.

I spent a couple hours to find out the root cause of an `OSError` when using S3FS which was caused by using aiobotocore with an incompatible version of botocore.
I'm using a Lambda Docker Image based on `public.ecr.aws/lambda/python:3.8` and I had to delete `/var/runtime/botocore` in order to make the Lambda use my provided botocore version.

In general, I would appreciate efforts to relax the version requirements on both boto3 and botocore because keeping everything working with the additional steps required in a use-case like mine may become a nightmare.
